### ISSUE TITLE
Force-Leave is not working : timed out broadcasting node removal from

### ISSUE THREAD

**Issue by shubhamitc** [hashicorp/consul#8179]
When filing a bug, please include the following headings if possible. Any example text in this template can be deleted.

#### Overview of the Issue

We are running our infra in EKS where consul clients are setup as daemon-sets. When a node is removed by cluster autoscaler, daemon-sets are not removed gracefully. The above condition makes the cluster report nodes in a failed state. We wrote a script based on EKS cluster events to capture RemovingNode event type, and mark the node force-leave. While executing the same we are getting an error: 
```
Jun 22 19:21:06 consul-xxxx-172 consul: 2020/06/22 19:21:06 [ERR] http: Request PUT /v1/agent/force-leave/ip-xxx.xxx.xxx.xxx.eu-west-1.compute.internal, error: timed out broadcasting node removal from=xxx.xxx.xxx.xxx:35096
```

Not sure if this is related but we are also seeing the below lines in consul logs: 
```
Jun 24 07:06:06 consul-goldendev-172 consul: 2020/06/24 07:06:06 [WARN] serf: Intent queue depth (13251) exceeds limit (4096), dropping messages!
```
#### Reproduction Steps

Steps to reproduce this issue, eg:

1. Create a cluster with 3 client nodes in EKS and 2 server nodes
1. Change the capacity from EKS to have 1 less node
1. Node will not go to left state, rather a failed state.
1. `curl --request PUT  http://172.28.31.77:8500/v1/agent/force-leave/"ip-172-28-30-194.eu-west-1.compute.internal"`

### Consul info for both Client and Server

<details>
  <summary>Client info</summary>

```
agent:

        check_monitors = 1

        check_ttls = 0

        checks = 1

        services = 1

build:

        prerelease =

        revision = 1200f25e

        version = 1.6.2

consul:

        acl = disabled

        known_servers = 3

        server = false

runtime:

        arch = amd64

        cpu_count = 2

        goroutines = 56

        max_procs = 2

        os = linux

        version = go1.12.13

serf_lan:

        coordinate_resets = 0

        encrypted = true

        event_queue = 0

        event_time = 7

        failed = 0

        health_score = 0

        intent_queue = 4863

        left = 25

        member_time = 31608

        members = 53

        query_queue = 0

        query_time = 10
```

</details>

<details>
  <summary>Server info</summary>

```
agent:

        check_monitors = 0

        check_ttls = 0

        checks = 0

        services = 0

build:

        prerelease =

        revision = 1200f25e

        version = 1.6.2

consul:

        acl = disabled

        bootstrap = false

        known_datacenters = 1

        leader = true

        leader_addr = 172.28.31.77:8300

        server = true

raft:

        applied_index = 3950734

        commit_index = 3950734

        fsm_pending = 0

        last_contact = 0

        last_log_index = 3950734

        last_log_term = 5235

        last_snapshot_index = 3935143

        last_snapshot_term = 5235

        latest_configuration = [{Suffrage:Voter ID:ca0ec93a-de5a-747e-02d4-736fe0041331 Address:172.28.31.77:8300} {Suffrage:Voter ID:30ba7e09-e173-2693-0b3b-5fe8aea8588c Address:172.28.30.34:8300} {Suffrage:Voter ID:44c28081-b10c-0bdc-95a9-dda4c6e7636c Address:172.28.30.60:8300}]

        latest_configuration_index = 7847

        num_peers = 2

        protocol_version = 3

        protocol_version_max = 3

        protocol_version_min = 0

        snapshot_version_max = 1

        snapshot_version_min = 0

        state = Leader

        term = 5235

runtime:

        arch = amd64

        cpu_count = 1

        goroutines = 191

        max_procs = 1

        os = linux

        version = go1.12.13

serf_lan:

        coordinate_resets = 0

        encrypted = true

        event_queue = 0

        event_time = 7

        failed = 0

        health_score = 0

        intent_queue = 11094

        left = 18

        member_time = 31608

        members = 46

        query_queue = 0

        query_time = 10

serf_wan:

        coordinate_resets = 0

        encrypted = true

        event_queue = 0

        event_time = 1

        failed = 0

        health_score = 0

        intent_queue = 0

        left = 0

        member_time = 266

        members = 3

        query_queue = 0

        query_time = 1```

</details>

### Operating system and Environment details

We are running servers on Amazon Linux V2 while clients are running on EKS as daemon-sets as well as on VM as a service. Some VMs are running older version of Amazon Linux V1 as well. 
### Log Fragments

Include appropriate Client or Server log fragments. If the log is longer than a few dozen lines, please include the URL to the [gist](https://gist.github.com/) of the log instead of posting it in the issue. Use `-log-level=TRACE` on the client and server to capture the maximum log detail.

**Comment by pierresouchay** [comment]
We have the same exact behavior sometimes when https://github.com/hashicorp/consul/issues/7960 triggers (same issue)

**Comment by t0klian** [comment]
We are observing the same behaviour of the Consul

**Comment by banks** [comment]
Hi,

We believe based on an investigation of a similar issue (not reported on GitHub) that we understand why this is happening and will be releasing a fix as soon as possible. More details to follow!

Thanks for reporting so clearly!

**Comment by rboyer** [comment]
This will be fixed and backported to consul 1.8.x, 1.7.x, and 1.6.x shortly. The details of the underlying bug can be found: https://github.com/hashicorp/serf/pull/606
